packages/frontend/src/app/repository/repository.component.spec.ts

```diff
@@ -0,0 +1,152 @@
+}
+
+@Component({
+  selector: 'app-filters',
+})
+class FiltersComponent {
+  setFilters = () => {};
+  @Input() set filters(filters) {}
+  @Output() filtersChanged = new EventEmitter<Filter[]>();
+}
+
+describe('RepositoryComponent', () => {
+  let component: RepositoryComponent;
+  let fixture: ComponentFixture<RepositoryComponent>;
+  let location: Location;
+
+  const getFilters = () => {
+    const filters = [];
+    let expectedRouteParams = '';
+    expectedParams.get('repo').filters.forEach(filterName => {
+      const filterValue = filterName + 'val';
+      filters.push({name: filterName, value: filterValue});
+      expectedRouteParams += ';' + filterName + '=' + filterValue;
+    });
+    return {filters, expectedRouteParams};
+  };
+
+  const getEnvironments = () => {
+    const {filters} = getFilters();
+    const environments = {};
+    filters.forEach(filter => {
+      environments[filter] = ['val1', 'val2'];
+    });
+    return environments;
+  };
+
+  // Mock services
+  const mockSearchService = {
+    searchBuilds: () => of({metadata: {environments: {}}}),
+  };
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      providers: [{provide: SearchService, useValue: mockSearchService}],
+      declarations: [
+        RepositoryComponent,
+        BuildListComponent,
+        SearchComponent,
+        FiltersComponent,
+      ],
+      imports: [AppRoutingModule, NoopAnimationsModule],
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(RepositoryComponent);
+    component = fixture.componentInstance;
+    fixture.autoDetectChanges(true);
+    mockSearchService.searchBuilds = () => of({metadata: {environments: {}}});
+    location = TestBed.get(Location);
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should set the new filters when a repository is found', done => {
+    const environments = getEnvironments();
+    component.filterComponent = {
+      setFilters: env => {
+        expect(env).toEqual(jasmine.objectContaining(environments));
+        done();
+      },
+    };
+
+    mockSearchService.searchBuilds = () =>
+      of({metadata: {environments: getEnvironments()}});
+
+    component.ngOnInit();
+  });
+
+  it('should redirect/refresh when the filters selection changes', fakeAsync(() => {
+    const filterComponent: FiltersComponent = fixture.debugElement.query(
+      By.css('app-filters')
+    ).componentInstance;
+
+    const {filters, expectedRouteParams} = getFilters();
+    const orgName = 'org',
+      repoName = 'repo';
+    const expectedRoute = '/' + orgName + '/' + repoName + expectedRouteParams;
+
+    component.orgName = orgName;
+    component.repoName = repoName;
+
+    filterComponent.filtersChanged.emit(filters);
+
+    tick();
+
+    expect(location.path()).toEqual(expectedRoute);
+  }));
+
+  it('should redirect to the 404 page if no repository was found', () => {
+    // TODO
```

Comment:
I believe this test case ends up getting filled in by the other PR?

Reply:
Yes. And I wanted to confirm that behavior. Do we want to redirect to the 404 or show a message instead (on the repo page).